docs/utilities/idempotency.md

```diff
@@ -0,0 +1,952 @@
+    
+        @BeforeAll
+        public static void setupDynamoLocal() {
+            int port = getFreePort();
+
+            // Initialize DynamoDBLocal
+            try {
+                DynamoDBProxyServer dynamoProxy = ServerRunner.createServerFromCommandLineArgs(new String[]{
+                        "-inMemory",
+                        "-port",
+                        Integer.toString(port)
+                });
+                dynamoProxy.start();
+            } catch (Exception e) {
+                throw new RuntimeException();
+            }
+    
+            // Initialize DynamoDBClient
+            client = DynamoDbClient.builder()
+                    .httpClient(UrlConnectionHttpClient.builder().build())
+                    .region(Region.EU_WEST_1)
+                    .endpointOverride(URI.create("http://localhost:" + port))
+                    .credentialsProvider(StaticCredentialsProvider.create(
+                            AwsBasicCredentials.create("FAKE", "FAKE")))
+                    .build();
+    
+            // create the table (use same table name as in pom.xml)
+            client.createTable(CreateTableRequest.builder()
+                    .tableName("idempotency")
+                    .keySchema(KeySchemaElement.builder().keyType(KeyType.HASH).attributeName("id").build())
+                    .attributeDefinitions(
+                            AttributeDefinition.builder().attributeName("id").attributeType(ScalarAttributeType.S).build()
+                    )
+                    .billingMode(BillingMode.PAY_PER_REQUEST)
+                    .build());
+        }
+    
+        private static int getFreePort() {
+            try {
+                ServerSocket socket = new ServerSocket(0);
+                int port = socket.getLocalPort();
+                socket.close();
+                return port;
+            } catch (IOException ioe) {
+                throw new RuntimeException(ioe);
+            }
+        }
+    
+        @BeforeEach
+        void setUp() {
+            MockitoAnnotations.openMocks(this);
+            app = new App(client);
+        }
+    
+        @Test
+        public void testApp() {
+            app.handleRequest(..., context);
+            // ... assert
+        }
+    }
+    ```
+
+=== "App.java"
+
+    ```java
+    public class App implements RequestHandler<Subscription, SubscriptionResult> {
+
+    public App(DynamoDbClient ddbClient) {
+        Idempotency.config().withPersistenceStore(
+                DynamoDBPersistenceStore.builder()
+                        .withTableName(System.getenv("IDEMPOTENCY_TABLE_NAME"))
+                        .withDynamoDbClient(ddbClient)
+                        .build()
+        ).configure();
+    }
+
+    public App() {
+        this(null);
+    }
+
+    @Idempotent
+    public SubscriptionResult handleRequest(final Subscription event, final Context context) {
+        // ...
+    }
+    ```
+
+!!! Warning
```

Comment:
We can probably remove this after dependency fix change is accepted ?

Reply:
Probably replace it with dependency recommendation instead if user is building on m1 mac ?

Reply:
added the dependency in the pom example